dataprofiler/tests/profilers/test_profile_builder.py

```diff
@@ -749,6 +752,417 @@ def test_min_max_id_properly_update(self, *mocks):
+        cls.profiler2 = UnstructuredProfiler(
+            pd.DataFrame(['extra', '\n', 'test\n', 'data .', 'For merging.']),
+            options=profiler_options
+        )
+        cls.report = cls.profiler.report()
+
+    def test_sample(self):
+        self.maxDiff = None
+        self.assertCountEqual(
+            ['Report',
+             'Removing duplication when you need a single source of truth is '
+             'almost always a win.',
+             'When you try to DRY them up into a single generic abstraction, '
+             'you have inadvertently coupled those two business rules '
+             'together.',
+             'edited 9 hours ago',
+             'Removing duplication that repeats the handling of the exact same '
+             'business rule is also usually a win.'],
+            self.profiler.sample
+        )
+
+    def test_total_samples(self):
+        self.assertEqual(20, self.profiler.total_samples)
+
+    def test_empty_line_count(self):
+        self.assertEqual(8, self.profiler._empty_line_count)
+
+    def test_text_profiler_results(self):
+        # pop out times
+        self.assertIsNotNone(
+            self.report['data_stats']['statistics'].pop('times'))
+
+        # vocab order doesn't matter
+        expected_vocab = ['x', 'i', 'y', 'q', 's', '9', ',', 'u', 'b', 'f', 'Y',
+                          'J', 'v', 'r', 'o', 'a', '6', 'n', 'T', 'k', 'h', ' ',
+                          'g', 'R', 't', 'W', '.', 'm', 'c', 'I', 'l', 'e', 'p',
+                          'w', 'S', 'd', 'D']
+        self.assertCountEqual(
+            expected_vocab,
+            self.report['data_stats']['statistics'].pop('vocab'))
+
+        # vocab order doesn't matter, case insensitive, remove stop words
+        expected_words = {
+            'Reply', 'also', 'a', 'function', 'DRY', 'creep', 'change',
+            'almost', 'modify', 'an', 'represent', '9', 'want', 'them', 'one',
+            'similar', 'different', 'caution', 'same', 'breaking',
+            'duplication', 'edited', 'in', 'top', 'that', 'together', 'has',
+            'thus', 'fit', 'and', 'source', 'rules', 'too', 'against',
+            'following', 'inadvertently', 'ago', 'requirements', 'duplicate',
+            'other', 'behavior', 'case', 'win', 'always', 'shared', 'single',
+            'on', 'have', 'repeats', 'but', 'whereby', 'lead', 'Removing',
+            'sometimes', 'code', 'the', '6', 'trying', 'those', 'up', 'it',
+            'business', 'two', 'rigidly', 'handling', 'truth', 'not', 'for',
+            'totally', 'is', 'part', 'hours', 'application', 'new', 'into',
+            'handle', 'rule', 'need', 'or', 'to', 'pieces', 'when', 'needs',
+            'can', 'handles', 'generic', 'Report', 'coupled', 'this', 'of',
+            'usually', 'try', 'if', 'just', 'special', 'Share', 'potential',
+            'your', 'unrelated', 'you', 'abstraction', 'exact', 'by',
+        }
```

Comment:
imo we should come up with less text, cause otherwise it's hard to manually inspect the results like this.

Reply:
done